packages/app/src/providers/hsds/utils.ts

```diff
@@ -115,37 +133,25 @@ export function convertHsdsType(hsdsType: HsdsType): DType {
       return convertHsdsCompoundType(hsdsType);
 
     case 'H5T_STRING':
-      return {
-        class: DTypeClass.String,
-        charSet: hsdsType.charSet.endsWith('ASCII') ? 'ASCII' : 'UTF-8',
-        length:
-          hsdsType.length === 'H5T_VARIABLE' ? undefined : hsdsType.length,
-      };
+      return strType(
+        hsdsType.charSet.endsWith('ASCII') ? 'ASCII' : 'UTF-8',
+        hsdsType.length === 'H5T_VARIABLE' ? undefined : hsdsType.length,
+      );
 
     case 'H5T_ARRAY':
     case 'H5T_VLEN':
       return {
         class:
           hsdsType.class === 'H5T_ARRAY' ? DTypeClass.Array : DTypeClass.VLen,
         base: convertHsdsType(hsdsType.base),
-        dims: hsdsType.dims,
+        ...(hsdsType.dims && { dims: hsdsType.dims }),
```

Comment:
We don't want `dims: undefined`. This was caught by switching from `toEqual` to `toStrictEqual` in the tests.

Reply:
:fishing_pole_and_fish: